simapp/simd/cmd/genaccounts.go

```diff
@@ -40,30 +39,32 @@ contain valid denominations. Accounts may optionally be supplied with vesting pa
 		Args: cobra.ExactArgs(2),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			clientCtx := client.GetClientContextFromCmd(cmd)
-			depCdc := clientCtx.Codec
-			cdc := depCdc.(codec.Codec)
 
 			serverCtx := server.GetServerContextFromCmd(cmd)
 			config := serverCtx.Config
 
 			config.SetRoot(clientCtx.HomeDir)
 
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			keyringBackend, _ := cmd.Flags().GetString(flags.FlagKeyringBackend)
+
+			var kr keyring.Keyring
 			addr, err := sdk.AccAddressFromBech32(args[0])
 			if err != nil {
-				inBuf := bufio.NewReader(cmd.InOrStdin())
-				keyringBackend, _ := cmd.Flags().GetString(flags.FlagKeyringBackend)
-
-				// attempt to lookup address from Keybase if no address was provided
-				kb, err := keyring.New(sdk.KeyringServiceName(), keyringBackend, clientCtx.HomeDir, inBuf)
-				if err != nil {
-					return err
+				if keyringBackend != "" && clientCtx.Keyring == nil {
+					var err error
+					kr, err = keyring.New(sdk.KeyringServiceName(), keyringBackend, clientCtx.HomeDir, inBuf)
+					if err != nil {
+						return err
+					}
```

Comment:
shouldn't we set:
```
clientCtx.Keyring = kr
```

Reply:
hmm im not sure i like the idea of overwriting clientCtx values

Reply:
What if for some reason we will use `clientCtx.Keyring` down the line?

Reply:
What if for some reason we will use `clientCtx.Keyring` down the line?

Reply:
the keyring in clientCtx comes from the user setting it in their config - if it needs to be used down the line i think it should be explicitly set by the user